src/validation.cpp

```diff
@@ -6108,23 +6127,25 @@ util::Result<void> ChainstateManager::PopulateAndValidateSnapshot(
 SnapshotCompletionResult ChainstateManager::MaybeCompleteSnapshotValidation()
 {
     AssertLockHeld(cs_main);
+
+    // If a snapshot does not need to be validated...
     if (m_ibd_chainstate.get() == &this->ActiveChainstate() ||
+            // Or if either chainstate is unusable...
             !this->IsUsable(m_snapshot_chainstate.get()) ||
             !this->IsUsable(m_ibd_chainstate.get()) ||
-            !m_ibd_chainstate->m_chain.Tip()) {
-       // Nothing to do - this function only applies to the background
-       // validation chainstate.
+            !m_snapshot_chainstate->m_from_snapshot_blockhash ||
```

Comment:
0cb19b78cc297ceca591ee8be6f63b4ea7b494cc: Can any of the conditions actually happen?

The snapshot chainstate not having base block hash set seems like a internal code error, than something that should be checked here? So an assert seems more appropriate, it at all? Same below for the other block hash checks.